devtools/gamma-server/test/gamma_server_and_cli_test.go

```diff
@@ -0,0 +1,276 @@
+		},
+	}
+
+	jsonBytes, _ := json.Marshal(&transferJSON)
+	return jsonBytes
+}
+
+func generateGetBalanceJSON(targetAddress []byte) []byte {
+	// Encode the address as hex
+	hexTargetAddress := hex.EncodeToString(targetAddress)
+
+	getBalanceJSON := &gammacli.JSONTransaction{
+		ContractName: "BenchmarkToken",
+		MethodName:   "getBalance",
+		Arguments: []gammacli.JSONMethodArgument{
+			{Name: "targetAddress", Type: "bytes", Value: hexTargetAddress},
+		},
+	}
+
+	callJSONBytes, _ := json.Marshal(&getBalanceJSON)
+	return callJSONBytes
+}
+
+func generateGetCounterJSON() []byte {
+	getCounterJSON := &gammacli.JSONTransaction{
+		ContractName: "Counter",
+		MethodName:   "get",
+		Arguments:    []gammacli.JSONMethodArgument{},
+	}
+
+	callJSONBytes, _ := json.Marshal(&getCounterJSON)
+	return callJSONBytes
+}
+
+func generateAddCounterJSON(amount uint64) []byte {
+	arg := gammacli.JSONMethodArgument{Name: "amount", Type: "uint64", Value: amount}
+
+	addAmountToCounterJSON := &gammacli.JSONTransaction{
+		ContractName: "Counter",
+		MethodName:   "add",
+		Arguments:    []gammacli.JSONMethodArgument{arg},
+	}
+
+	addJSONBytes, _ := json.Marshal(&addAmountToCounterJSON)
+	return addJSONBytes
+}
+
+func TestGammaFlowWithActualJSONFilesUsingBenchmarkToken(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping e2e tests in short mode")
+	}
+
+	port := ":8080"
+	gamma := gammacli.StartGammaServer(port, false)
+	defer gamma.GracefulShutdown(1 * time.Second)
+
+	time.Sleep(100 * time.Millisecond) // wait for server to start
+
+	keyPair := keys.Ed25519KeyPairForTests(0)
+	targetAddress := builders.AddressForEd25519SignerForTests(2)
+
+	transferJSONBytes := generateTransferJSON(42, targetAddress)
+
+	err := ioutil.WriteFile("../json/transfer.json", transferJSONBytes, 0644)
+	if err != nil {
+		fmt.Println("Couldn't write file", err)
+	}
+	require.NoError(t, err, "Couldn't write transfer JSON file")
+
+	baseCommand := ClientBinary()
+	sendCommand := append(baseCommand,
+		"run", "send", "../json/transfer.json",
+		"-public-key", keyPair.PublicKey().String(),
+		"-private-key", keyPair.PrivateKey().String())
+
+	sendCommandOutput := runCommand(sendCommand, t)
+
+	response := &sendTransactionCliResponse{}
+	unmarshalErr := json.Unmarshal([]byte(sendCommandOutput), &response)
+
+	require.NoError(t, unmarshalErr, "error unmarshal cli response")
+	require.Equal(t, 1, response.TransactionReceipt.ExecutionResult, "JSONTransaction status to be successful = 1")
+	require.Equal(t, 1, response.TransactionStatus, "JSONTransaction status to be successful = 1")
+	require.NotNil(t, response.TransactionReceipt.Txhash, "got empty txhash")
+
+	getBalanceJSONBytes := generateGetBalanceJSON(targetAddress)
+	err = ioutil.WriteFile("../json/getBalance.json", getBalanceJSONBytes, 0644)
+	if err != nil {
+		fmt.Println("Couldn't write file", err)
+	}
+	require.NoError(t, err, "Couldn't write getBalance JSON file")
```

Comment:
I think the test should panic if `err!=nil`
Also, what is the working dir? where are you writing the file to? are you polluting my user dir with your json files?

Reply:
Basically the directory that the JSONs are written to is the one relative to the test folder, there's a directory aside to it which is included into the .gitignore and ignores all JSONS written to that folder.